### PR TITLE
Added episode generation code for PointGoal task.

### PR DESCRIPTION
**Pull request by mathfac**
## Motivation and Context
- Added episode generation code for PointGoal task.
- Added generator test with shortest path proper test and episode serialization testing. 
- Fixed formatting issues and unused imports across the codebase.
- Added editor config style formatter properties.

## How Has This Been Tested
`pytest`
`black . --line-length 79;mypy . --ignore-missing-imports;pycodestyle;flake8`
```
============================================================================================================ test session starts =============================================================================================================
platform linux -- Python 3.6.4, pytest-4.4.1, py-1.5.2, pluggy-0.9.0
rootdir: /private/home/maksymets/habitat-api, inifile: setup.cfg, testpaths: test
plugins: xdist-1.25.0, forked-0.2
collected 37 items

test/test_baseline_agents.py ..                                                                                                                                                                                                        [  5%]
test/test_config.py .                                                                                                                                                                                                                  [  8%]
test/test_dataset.py .........                                                                                                                                                                                                         [ 32%]
test/test_habitat_env.py .........s                                                                                                                                                                                                    [ 59%]
test/test_habitat_example.py ...                                                                                                                                                                                                       [ 67%]
test/test_install.py .                                                                                                                                                                                                                 [ 70%]
test/test_mp3d_eqa.py ...                                                                                                                                                                                                              [ 78%]
test/test_pointnav_dataset.py ...                                                                                                                                                                                                      [ 86%]
test/test_sensors.py ...                                                                                                                                                                                                               [ 94%]
test/test_trajectory_sim.py ..                                                                                                                                                                                                         [100%]
========================================================================================================== short test summary info ===========================================================================================================
SKIPPED [1] test/test_habitat_env.py:318: unconditional skip

=================================================================================================== 36 passed, 1 skipped in 210.82 seconds ===================================================================================================
```
<!--- Please describe here how your modifications have been tested. -->

## Types of changes

<!--- What types of changes does your code introduce? Leave all the items that apply: -->
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: habitat/datasets/pointnav/generator.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+            is_compatible, dist = is_compatible_episode(
+                source_position,
+                target_position,
+                env,
+                near_dist=NEAR_DIST_LIMIT,
+                far_dist=FAR_DIST_LIMIT,
+            )
+            if is_compatible:
+                angle = np.random.uniform(0, 2 * np.pi)
+                source_rotation = [0, np.sin(angle / 2), 0, np.cos(angle / 2)]
```
Comment (abhiskk) [inline habitat/datasets/pointnav/generator.py]: Should this be `[0, np.sin(angle), 0, np.cos(angle)]` ?

Comment (abhiskk) [reply habitat/datasets/pointnav/generator.py]: This is correct, `sin(angle / 2)` and `cos(angle / 2)` uniquely identify the `angle`. We should use a utility function for this, I think there is an `angle` to `quaternion` utility function which we can call here.

Comment (abhiskk) [reply habitat/datasets/pointnav/generator.py]: Here is a visualization to confirm this: 
image

As the graph never intersects itself each point is uniquely identified.

Comment (erikwijmans) [reply habitat/datasets/pointnav/generator.py]: You can use something like `quat_from_angle_axis` for this https://github.com/facebookresearch/habitat-sim/blob/master/habitat_sim/utils.py#L66

In this case, the axis would be the up direction, `[0, 1, 0]` and the angle would be `theta \in [0, 2 PI]`

File: habitat/datasets/pointnav/generator.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    """
+
+    :param ratio: geodesic distance ratio to Euclid distance
+    :return: value between 0.008 and 0.144 for ration 1 and 1.1
+    """
+    return 20 * (ratio - 0.98) ** 2
+
+
+def is_compatible_episode(s, t, env, near_dist, far_dist):
+    substract_vec = [a_i - b_i for a_i, b_i in zip(s, t)]
```
Comment (erikwijmans) [inline habitat/datasets/pointnav/generator.py]: Why not just `s - t` or `np.array(s) - np.array(t)`?

Comment (mathfac) [reply habitat/datasets/pointnav/generator.py]: @erikwijmans, `s - t` won't work on python lists, while `np.array(s) - np.array(t)` is good idea.

Comment (erikwijmans) [reply habitat/datasets/pointnav/generator.py]: `s` and `t` are gotten from `sample_navigable_point()` which returns `np.array`s

Comment (mathfac) [reply habitat/datasets/pointnav/generator.py]: In HAPI sim wrapper we convert to list already: https://github.com/facebookresearch/habitat-api/blob/8c5329ce9f777395b45c2daac135377e629ec148/habitat/sims/habitat_simulator.py#L338. Created issue https://github.com/facebookresearch/habitat-api/issues/84 to start moving all HAPI interfaces to np.array.

**Comment by mathfac** [comment]
@abhiskk, @erikwijmans final look required.

File: habitat/tasks/utils.py [hunk 3]
```diff
@@ -5,7 +5,6 @@
 # LICENSE file in the root directory of this source tree.
 
 import numpy as np
-import quaternion
```
Comment (erikwijmans) [inline habitat/tasks/utils.py]: This import should stay as you can't be sure that numpy will be patched and `np.quaternion` will exist otherwise.

Comment (mathfac) [reply habitat/tasks/utils.py]: Nice catch!.

**Pull request merged**